Source/WebKit/Platform/IPC/MessageSenderInlines.h

```diff
@@ -158,4 +158,22 @@ template<typename MessageType, typename C, typename U, typename V> inline AsyncR
     return sendWithAsyncReply(std::forward<MessageType>(message), std::forward<C>(completionHandler), destinationID.toUInt64(), options);
 }
 
+template<typename MessageType> Ref<typename MessageType::Promise> inline MessageSender::sendWithPromisedReply(MessageType&& message)
+{
+    return sendWithPromisedReply(std::forward<MessageType>(message), messageSenderDestinationID(), { });
+}
+
+template<typename MessageType> Ref<typename MessageType::Promise> inline MessageSender::sendWithPromisedReply(MessageType&& message, uint64_t destinationID)
+{
+    return sendWithPromisedReply(std::forward<MessageType>(message), destinationID, { });
+}
+
+template<typename MessageType> Ref<typename MessageType::Promise> inline MessageSender::sendWithPromisedReply(MessageType&& message, uint64_t destinationID, OptionSet<SendOption> options)
+{
+    static_assert(!MessageType::isSync);
+    if (auto* connection = messageSenderConnection())
```

Comment:
s/auto*/RefPtr

Reply:
The code is all inlined here and it’s not doing reference churn above either. 
do we want/need to do that here?

Reply:
`Implicit instantiation of undefined template 'WTF::OptionSet<IPC::SendOption>'`

That doesn't work with the inline; and considering this was all done to speed-up compilation times, I'll leave it as-is